File: 0252-meeting-rooms/0252-meeting-rooms.cpp
// TC- O(nlogn)
// SC- O(logn)

class Solution {
public:
    bool canAttendMeetings(vector<vector<int>>& intervals) {
        if(intervals.size()==0)
            return true;

        sort(intervals.begin(), intervals.end(), [](vector<int>& a, vector<int>& b){return a[0]<b[0];});
        for(int i=0; i<intervals.size()-1; i++){
            if(intervals[i+1][0]<intervals[i][1])
                return false;
        }
        return true;
    }
};

File: 0252-meeting-rooms/0252-meeting-rooms.py
# TC- O(nlogn)
# SC- O(n)

class Solution:
    def canAttendMeetings(self, intervals: List[List[int]]) -> bool:
        intervals.sort()

        for i in range(1, len(intervals)):
            if(intervals[i][0]<intervals[i-1][1]):
                return False
        return True